# 31. Macro Constraint: The Demographic Winter (*Denatalità*)
This notebook models the demographic collapse in Italy based on ISTAT projections to demonstrate how the shrinking student population affects school consolidation, funding, and the physical survival of the education infrastructure, particularly in the South.

In [1]:
import pandas as pd
import os

os.makedirs('local_data/processed/macro_constraints', exist_ok=True)

### 1. ISTAT Population Projections (Student Age 15-19)
We model the projected collapse of the upper-secondary school population from 2025 through 2040.

In [2]:
demographic_data = {
    'Year': [2020, 2025, 2030, 2035, 2040],
    'Student_Population_15_19_Millions': [2.85, 2.72, 2.45, 2.10, 1.95], # ISTAT projection models
    'Projected_School_Closures_Pct': [0, 4.5, 12.0, 22.5, 30.0] # Schools mathematically forced to consolidate due to size
}
df_demographics = pd.DataFrame(demographic_data)
df_demographics.to_csv('local_data/processed/macro_constraints/istat_demographic_winter.csv', index=False)
df_demographics

,Year,Student_Population_15_19_Millions,Projected_School_Closures_Pct
0,2020,2.85,0.0
1,2025,2.72,4.5
2,2030,2.45,12.0
3,2035,2.10,22.5
4,2040,1.95,30.0


### 2. Analysis
The *Denatalità* acts as an accelerating constraint on the education system. As the student population drops from 2.8 million to under 2.0 million by 2040, up to 30% of physical school locations (plexuses) face closure or consolidation.

This disproportionately hits underdeveloped areas in the South and Mountains, forcing longer commute times and destroying the "Origin" communities.